src/internal/methods/PixieSerialMethod.h

```diff
@@ -0,0 +1,99 @@
+/*-------------------------------------------------------------------------
+NeoPixel library helper functions for DotStars using general Pins (APA102).
+
+Written by Michael C. Miller.
+
+I invest time and resources providing this open source code,
+please support me by dontating (see https://github.com/Makuna/NeoPixelBus)
+
+-------------------------------------------------------------------------
+This file is part of the Makuna/NeoPixelBus library.
+
+NeoPixelBus is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as
+published by the Free Software Foundation, either version 3 of
+the License, or (at your option) any later version.
+
+NeoPixelBus is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with NeoPixel.  If not, see
+<http://www.gnu.org/licenses/>.
+-------------------------------------------------------------------------*/
+
+/*
+  Pixie reads data in at 115.2k serial, 8N1.
+  Byte order is R1, G1, B1, R2, G2, B2, ... where the first triplet is the
+  color of the LED that's closest to the controller. 1ms of silence triggers
+  latch. 2 seconds silence (or overheating) triggers LED off (for safety).
+*/
+
+
+#pragma once
+
+
+class PixieSerialMethod
+{
+public:
+    typedef NeoNoSettings SettingsObject;
+
+    PixieSerialMethod(uint16_t pixelCount, size_t elementSize, size_t settingsSize, Stream *s) :
+        _sizeData(pixelCount * elementSize + settingsSize),
+        _endTime(0),
+        _stream(s)
+    {
+        _data = static_cast<uint8_t*>(malloc(_sizeData));
+        // data cleared later in Begin()
+    }
+
+    ~PixieSerialMethod()
+    {
+        free(_data);
+    }
+
+    bool IsReadyToUpdate() const
+    {
+        return (micros() - _endTime) > 1000L; // ensure 1ms delay between calls
+    }
+
+    void Initialize()
+    {
+        //nothing to initialize, UART is managed outside this library
+    }
+
+    void Update(bool)
+    {
+        _stream->write(_data, _sizeData);
+        _endTime = micros(); // Save time to ensure 1ms delay
+    }
+
+    bool AlwaysUpdate()
+    {
+        //Pixie expects to receive data every <2 seconds
+        return true;
```

Comment:
You maybe confusing what AlwaysUpdate is used for.  It is a means for the method to inform layers above that the use of a dirty flag and only update when dirty flag is set or in this case, ignore the dirty flag and always update.  I think returning true is still correct here, but the comment isn't detailed enough to make this clear.

Reply:
Yes, I believe it should be true as well.  I updated my comment to make it more clear and describe the auto-off condition.  Let me know if this is still not clear enough.

// Pixie expects to receive data every <2 seconds, Adafruit recommends <1.  
// Ensuring data is sent every <2 seconds needs to happen outside of the library
// Returning true will allow data to be re-sent even if no changes to buffer.